integration/setup/bastion_instance.go

```diff
@@ -0,0 +1,559 @@
+	// That is why we need to separately create tags below, so we are able to find
+	// it later on when we want to clean it up.
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "tagging bastion security group")
+
+		i := &ec2.CreateTagsInput{
+			Resources: []*string{
+				aws.String(bastionSecurityGroupID),
+			},
+			Tags: []*ec2.Tag{
+				{
+					Key:   aws.String("Name"),
+					Value: aws.String(clusterID + "-bastion"),
+				},
+				{
+					Key:   aws.String("giantswarm.io/cluster"),
+					Value: aws.String(clusterID),
+				},
+			},
+		}
+
+		_, err = config.AWSClient.EC2.CreateTags(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "tagged bastion security group")
+	}
+
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "updating bastion security group to allow ssh access")
+
+		i := &ec2.AuthorizeSecurityGroupIngressInput{
+			GroupId: aws.String(bastionSecurityGroupID),
+			IpPermissions: []*ec2.IpPermission{
+				{
+					FromPort:   aws.Int64(22),
+					IpProtocol: aws.String("tcp"),
+					IpRanges: []*ec2.IpRange{
+						{
+							CidrIp:      aws.String("0.0.0.0/0"),
+							Description: aws.String("Allow SSH access from everywhere to port 22."),
+						},
+					},
+					ToPort: aws.Int64(22),
+				},
+			},
+		}
+
+		_, err = config.AWSClient.EC2.AuthorizeSecurityGroupIngress(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "updated bastion security group to allow ssh access")
+	}
+
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "creating bastion instance")
+
+		i := &ec2.RunInstancesInput{
+			ImageId:      aws.String("ami-015e6cb33a709348e"),
+			InstanceType: aws.String("t2.micro"),
+			MaxCount:     aws.Int64(1),
+			MinCount:     aws.Int64(1),
+			NetworkInterfaces: []*ec2.InstanceNetworkInterfaceSpecification{
+				{
+					AssociatePublicIpAddress: aws.Bool(true),
+					DeviceIndex:              aws.Int64(0),
+					Groups: []*string{
+						aws.String(bastionSecurityGroupID),
+						aws.String(workerSecurityGroupID),
+					},
+					SubnetId: aws.String(subnetID),
+				},
+			},
+			TagSpecifications: []*ec2.TagSpecification{
+				{
+					ResourceType: aws.String("instance"),
+					Tags: []*ec2.Tag{
+						{
+							Key:   aws.String("Name"),
+							Value: aws.String(clusterID + "-bastion"),
+						},
+						{
+							Key:   aws.String("giantswarm.io/cluster"),
+							Value: aws.String(clusterID),
+						},
+						{
+							Key:   aws.String("giantswarm.io/instance"),
+							Value: aws.String("e2e-bastion"),
+						},
+					},
+				},
+			},
+			UserData: aws.String(base64.StdEncoding.EncodeToString([]byte(`
+				{
+				  "ignition": {
+				    "config": {},
+				    "timeouts": {},
+				    "version": "2.1.0"
+				  },
+				  "networkd": {},
+				  "passwd": {
+				    "users": [
```

Comment:
From a quick look it seems that all the new guys are missing from here. Salvo, Ferran, Ji etc. Maybe ping them after this is merged to add their keys in order to get access?

Reply:
@mazzy89 @ferrandinand @tomahawk28 your keys are not in the installations repo and thus are missing here. Please make sure your keys are available so you can access tenant clusters. Then also make sure you add yourself here in case you are interested in debugging tenants within e2e tests.

Reply:
It does not feels right to me to have all those ssh keys hardcoded here. We should have a central place for this. It makes it hard later on to grant a new user access to all our infrastructure.

Reply:
I agree. A better solution would be to fetch it from the installations repo. We do the same with the version bundle versions. I would just like to not do this now. I will add this note to the TODO issue we created earlier. Thanks for pointing out.